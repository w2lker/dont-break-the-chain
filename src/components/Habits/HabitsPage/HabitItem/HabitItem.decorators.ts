import withStyles from '@material-ui/core/styles/withStyles';
import HabitItem from './HabitItem';
import habitItemStyles from './HabitItem.styles';

const styledComponent = withStyles(habitItemStyles)(HabitItem);

export default styledComponent;
