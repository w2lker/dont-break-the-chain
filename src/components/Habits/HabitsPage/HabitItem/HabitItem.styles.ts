const habitItemStyles = {
  root: {
    padding: 0,
  },
};

export default habitItemStyles;
