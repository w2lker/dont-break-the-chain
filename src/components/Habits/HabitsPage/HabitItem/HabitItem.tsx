import React from 'react';

import { WithStyles } from '@material-ui/styles';

import habitItemStyles from './HabitItem.styles';

export interface IHabitItemProps extends WithStyles<typeof habitItemStyles> {}

const HabitItem: React.FC<IHabitItemProps> = (props) => {
  const { classes } = props;
  return (
    <div className={classes.root}>
      HabitItem Component
    </div>
  );
};

export default HabitItem;
