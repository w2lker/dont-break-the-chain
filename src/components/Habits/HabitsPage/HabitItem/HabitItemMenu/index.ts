import HabitItemMenu from './HabitItemMenu.decorators';

export default HabitItemMenu;
