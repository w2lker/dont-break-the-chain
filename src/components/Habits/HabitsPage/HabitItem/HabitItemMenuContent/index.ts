import HabitItemMenuContent from './HabitItemMenuContent.decorators';

export default HabitItemMenuContent;
