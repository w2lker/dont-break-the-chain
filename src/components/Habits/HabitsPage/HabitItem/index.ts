import HabitItem from './HabitItem.decorators';

export default HabitItem;
