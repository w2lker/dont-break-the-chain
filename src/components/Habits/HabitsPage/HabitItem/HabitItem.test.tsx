import { List } from 'immutable';
import React from 'react';

import { mount, shallow } from 'enzyme';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import { MemoryRouter } from 'react-router-dom';
import HabitItem, { IHabitItemProps } from './HabitItem';
import { habitItemStyles } from './HabitItem.styles';

const middleware = [thunk];
const mockStore = configureMockStore(middleware);

const testSetup = () => {
  const classes = {
    wrapper: 'habitItem-wrapper',
    lastWrapper: 'habitItem-last-wrapper',
    icon: 'habitItem-icon',
    checkMark: 'habitItem-check-mark',
    description: 'habitItem-description',
    title: 'habitItem-title',
    subtitle: 'habitItem-subtitle',
  };

  const classesKeys = Object.keys(classes);

  const sampleProps: IHabitItemProps = {
    content: {
      id: 1,
      name: 'Sample habit',
      color: 'blue',
      icon: 'sample-icon',
      currentChain: 1,
      longestChain: 2,
      dates: List(),
    },
    isLast: false,
  };

  const store = mockStore({});

  const wrappedComponent = (props: IHabitItemProps) => (
    <MemoryRouter>
      <Provider store={store}>
        <HabitItem {...props}/>
      </Provider>
    </MemoryRouter>
  );

  return { classes, classesKeys, sampleProps, wrappedComponent };
};

describe('HabitItem component', () => {

  it('match snapshot', () => {
    const { sampleProps, wrappedComponent } = testSetup();
    const component = mount(wrappedComponent(sampleProps));
    expect(component.debug()).toMatchSnapshot();
  });

  it('skips render with incorrect habit', () => {
    const { sampleProps, wrappedComponent } = testSetup();
    const props = {
      ...sampleProps,
      content: {
        ...sampleProps.content,
        name: '',
      },
    };
    const component = shallow(<HabitItem {...props} />);
    expect(component.type()).toEqual(null);
  });
});

describe('HabitItem styles', () => {
  it('contains principle classes', () => {
    const { classesKeys } = testSetup();
    expect(habitItemStyles).toBeDefined();
    classesKeys.forEach((keyValue) => {
      // @ts-ignore
      expect(habitItemStyles[keyValue]).toBeDefined();
    });
  });
});
