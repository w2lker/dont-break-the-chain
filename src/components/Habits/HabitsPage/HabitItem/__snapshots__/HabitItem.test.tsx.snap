// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HabitItem component match snapshot 1`] = `
"<MemoryRouter>
  <Router history={{...}}>
    <Provider store={{...}}>
      <HabitItem content={{...}} isLast={false}>
        <div className=\\"wrapper-0-2-1\\">
          <div className=\\"icon-0-2-3 icon-0-0-2-8\\">
            <MaterialDesignIcon name=\\"sample-icon\\">
              <span className=\\"material-icon mdi mdi-sample-icon\\" style={[undefined]} />
            </MaterialDesignIcon>
          </div>
          <div className=\\"description-0-2-5\\">
            <h4 className=\\"title-0-2-6\\">
              Sample habit
            </h4>
            <span className=\\"subtitle-0-2-7\\">
              Today - incomplete
            </span>
            <span className=\\"subtitle-0-2-7\\">
              Chain - 1 days
            </span>
            <WithStyles(undefined) content={{...}}>
              <Connect(HabitItemMenu) classes={{...}} content={{...}}>
                <HabitItemMenu classes={{...}} content={{...}} onCheck={[Function: onCheck]} onUncheck={[Function: onUncheck]} onPause={[Function: onPause]} onDelete={[Function: onDelete]}>
                  <div className=\\"Connect(HabitItemMenu)-wrapper-1\\">
                    <WithStyles(ForwardRef(IconButton)) onClick={[Function: onClick]}>
                      <ForwardRef(IconButton) classes={{...}} onClick={[Function: onClick]}>
                        <WithStyles(ForwardRef(ButtonBase)) className=\\"MuiIconButton-root\\" centerRipple={true} focusRipple={true} disabled={false} onClick={[Function: onClick]}>
                          <ForwardRef(ButtonBase) classes={{...}} className=\\"MuiIconButton-root\\" centerRipple={true} focusRipple={true} disabled={false} onClick={[Function: onClick]}>
                            <button className=\\"MuiButtonBase-root MuiIconButton-root\\" onBlur={[Function]} onClick={[Function: onClick]} onFocus={[Function]} onKeyDown={[Function]} onKeyUp={[Function]} onMouseDown={[Function]} onMouseLeave={[Function]} onMouseUp={[Function]} onDragLeave={[Function]} onTouchEnd={[Function]} onTouchMove={[Function]} onTouchStart={[Function]} tabIndex={0} type=\\"button\\" disabled={false}>
                              <span className=\\"MuiIconButton-label\\">
                                <MaterialDesignIcon name=\\"dots-vertical\\">
                                  <span className=\\"material-icon mdi mdi-dots-vertical\\" style={[undefined]} />
                                </MaterialDesignIcon>
                              </span>
                              <NoSsr>
                                <WithStyles(undefined) center={true}>
                                  <ForwardRef(TouchRipple) classes={{...}} center={true}>
                                    <span className=\\"MuiTouchRipple-root\\">
                                      <TransitionGroup component={{...}} exit={true} childFactory={[Function: childFactory]} />
                                    </span>
                                  </ForwardRef(TouchRipple)>
                                </WithStyles(undefined)>
                              </NoSsr>
                            </button>
                          </ForwardRef(ButtonBase)>
                        </WithStyles(ForwardRef(ButtonBase))>
                      </ForwardRef(IconButton)>
                    </WithStyles(ForwardRef(IconButton))>
                    <WithStyles(ForwardRef(Menu)) id=\\"habit-menu-1\\" anchorEl={{...}} keepMounted={true} open={false} onClose={[Function: onClose]}>
                      <ForwardRef(Menu) classes={{...}} id=\\"habit-menu-1\\" anchorEl={{...}} keepMounted={true} open={false} onClose={[Function: onClose]}>
                        <WithStyles(ForwardRef(Popover)) getContentAnchorEl={[Function: getContentAnchorEl]} classes={[undefined]} onClose={[Function: onClose]} onEntering={[Function: handleEntering]} anchorOrigin={{...}} transformOrigin={{...}} PaperProps={{...}} open={false} transitionDuration=\\"auto\\" id=\\"habit-menu-1\\" anchorEl={{...}} keepMounted={true}>
                          <ForwardRef(Popover) classes={{...}} getContentAnchorEl={[Function: getContentAnchorEl]} onClose={[Function: onClose]} onEntering={[Function: handleEntering]} anchorOrigin={{...}} transformOrigin={{...}} PaperProps={{...}} open={false} transitionDuration=\\"auto\\" id=\\"habit-menu-1\\" anchorEl={{...}} keepMounted={true}>
                            <ForwardRef(Modal) container={[undefined]} open={false} BackdropProps={{...}} className=\\"MuiPopover-root\\" onClose={[Function: onClose]} id=\\"habit-menu-1\\" keepMounted={true}>
                              <ForwardRef(Portal) container={[undefined]} disablePortal={false}>
                                <Portal containerInfo={{...}}>
                                  <div onKeyDown={[Function: handleKeyDown]} role=\\"presentation\\" className=\\"MuiPopover-root\\" id=\\"habit-menu-1\\" style={{...}}>
                                    <ForwardRef(SimpleBackdrop) open={false} onClick={[Function: handleBackdropClick]} invisible={true} />
                                    <TrapFocus disableEnforceFocus={false} disableAutoFocus={false} disableRestoreFocus={false} getDoc={[Function: getDoc]} isEnabled={[Function]} open={false}>
                                      <div tabIndex={0} data-test=\\"sentinelStart\\" />
                                      <ForwardRef(Grow) appear={true} in={false} onEnter={[Function: chainedFunction]} onEntered={[undefined]} onExit={[undefined]} onExited={[Function: chainedFunction]} onExiting={[undefined]} timeout=\\"auto\\" onEntering={[Function: chainedFunction]} tabIndex=\\"-1\\">
                                        <Transition appear={true} in={false} onEnter={[Function: handleEnter]} onExit={[Function: handleExit]} addEndListener={[Function: addEndListener]} timeout={{...}} onEntered={[Function: noop]} onExited={[Function: chainedFunction]} onExiting={[Function: noop]} onEntering={[Function: chainedFunction]} tabIndex=\\"-1\\" mountOnEnter={false} unmountOnExit={false} enter={true} exit={true}>
                                          <WithStyles(ForwardRef(Paper)) elevation={8} classes={{...}} className=\\"MuiPopover-paper\\" style={{...}} tabIndex=\\"-1\\">
                                            <ForwardRef(Paper) classes={{...}} elevation={8} className=\\"MuiPopover-paper\\" style={{...}} tabIndex=\\"-1\\">
                                              <div className=\\"MuiPaper-root MuiMenu-paper MuiPaper-elevation8 MuiPopover-paper MuiPaper-rounded\\" style={{...}} tabIndex=\\"-1\\">
                                                <ForwardRef(MenuList) onKeyDown={[Function: handleListKeyDown]} actions={{...}} autoFocus={false} autoFocusItem={false} variant=\\"selectedMenu\\" className=\\"MuiMenu-list\\">
                                                  <WithStyles(ForwardRef(List)) role=\\"menu\\" className=\\"MuiMenu-list\\" onKeyDown={[Function: handleKeyDown]} tabIndex={-1}>
                                                    <ForwardRef(List) classes={{...}} role=\\"menu\\" className=\\"MuiMenu-list\\" onKeyDown={[Function: handleKeyDown]} tabIndex={-1}>
                                                      <ul className=\\"MuiList-root MuiMenu-list MuiList-padding\\" role=\\"menu\\" onKeyDown={[Function: handleKeyDown]} tabIndex={-1}>
                                                        <WithStyles(HabitItemMenuContent) icon=\\"check\\" title=\\"Complete today\\" action={[Function: onCheck]} tabIndex={0}>
                                                          <HabitItemMenuContent classes={{...}} icon=\\"check\\" title=\\"Complete today\\" action={[Function: onCheck]} tabIndex={0}>
                                                            <WithStyles(ForwardRef(MenuItem)) onClick={[Function: onCheck]}>
                                                              <ForwardRef(MenuItem) classes={{...}} onClick={[Function: onCheck]}>
                                                                <WithStyles(ForwardRef(ListItem)) button={true} role=\\"menuitem\\" tabIndex={-1} component=\\"li\\" selected={[undefined]} disableGutters={false} classes={{...}} className=\\"MuiMenuItem-root MuiMenuItem-gutters\\" onClick={[Function: onCheck]}>
                                                                  <ForwardRef(ListItem) classes={{...}} button={true} role=\\"menuitem\\" tabIndex={-1} component=\\"li\\" selected={[undefined]} disableGutters={false} className=\\"MuiMenuItem-root MuiMenuItem-gutters\\" onClick={[Function: onCheck]}>
                                                                    <WithStyles(ForwardRef(ButtonBase)) className=\\"MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button\\" disabled={false} role=\\"menuitem\\" tabIndex={-1} onClick={[Function: onCheck]} component=\\"li\\" focusVisibleClassName=\\"Mui-focusVisible\\">
                                                                      <ForwardRef(ButtonBase) classes={{...}} className=\\"MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button\\" disabled={false} role=\\"menuitem\\" tabIndex={-1} onClick={[Function: onCheck]} component=\\"li\\" focusVisibleClassName=\\"Mui-focusVisible\\">
                                                                        <li className=\\"MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button\\" onBlur={[Function]} onClick={[Function: onCheck]} onFocus={[Function]} onKeyDown={[Function]} onKeyUp={[Function]} onMouseDown={[Function]} onMouseLeave={[Function]} onMouseUp={[Function]} onDragLeave={[Function]} onTouchEnd={[Function]} onTouchMove={[Function]} onTouchStart={[Function]} tabIndex={-1} role=\\"menuitem\\" aria-disabled={false}>
                                                                          <div className=\\"HabitItemMenuContent-root-59\\">
                                                                            <MaterialDesignIcon className=\\"HabitItemMenuContent-icon-60\\" name=\\"check\\">
                                                                              <span className=\\"material-icon mdi mdi-check HabitItemMenuContent-icon-60\\" style={[undefined]} />
                                                                            </MaterialDesignIcon>
                                                                            <span className=\\"HabitItemMenuContent-text-61\\">
                                                                              Complete today
                                                                            </span>
                                                                          </div>
                                                                          <NoSsr>
                                                                            <WithStyles(undefined) center={false}>
                                                                              <ForwardRef(TouchRipple) classes={{...}} center={false}>
                                                                                <span className=\\"MuiTouchRipple-root\\">
                                                                                  <TransitionGroup component={{...}} exit={true} childFactory={[Function: childFactory]} />
                                                                                </span>
                                                                              </ForwardRef(TouchRipple)>
                                                                            </WithStyles(undefined)>
                                                                          </NoSsr>
                                                                        </li>
                                                                      </ForwardRef(ButtonBase)>
                                                                    </WithStyles(ForwardRef(ButtonBase))>
                                                                  </ForwardRef(ListItem)>
                                                                </WithStyles(ForwardRef(ListItem))>
                                                              </ForwardRef(MenuItem)>
                                                            </WithStyles(ForwardRef(MenuItem))>
                                                          </HabitItemMenuContent>
                                                        </WithStyles(HabitItemMenuContent)>
                                                        <WithStyles(HabitItemMenuContent) icon=\\"pencil\\" title=\\"Edit\\" url=\\"/habitid1\\">
                                                          <HabitItemMenuContent classes={{...}} icon=\\"pencil\\" title=\\"Edit\\" url=\\"/habitid1\\">
                                                            <WithStyles(ForwardRef(MenuItem)) component={{...}} to=\\"/habitid1\\">
                                                              <ForwardRef(MenuItem) classes={{...}} component={{...}} to=\\"/habitid1\\">
                                                                <WithStyles(ForwardRef(ListItem)) button={true} role=\\"menuitem\\" tabIndex={-1} component={{...}} selected={[undefined]} disableGutters={false} classes={{...}} className=\\"MuiMenuItem-root MuiMenuItem-gutters\\" to=\\"/habitid1\\">
                                                                  <ForwardRef(ListItem) classes={{...}} button={true} role=\\"menuitem\\" tabIndex={-1} component={{...}} selected={[undefined]} disableGutters={false} className=\\"MuiMenuItem-root MuiMenuItem-gutters\\" to=\\"/habitid1\\">
                                                                    <WithStyles(ForwardRef(ButtonBase)) className=\\"MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button\\" disabled={false} role=\\"menuitem\\" tabIndex={-1} to=\\"/habitid1\\" component={{...}} focusVisibleClassName=\\"Mui-focusVisible\\">
                                                                      <ForwardRef(ButtonBase) classes={{...}} className=\\"MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button\\" disabled={false} role=\\"menuitem\\" tabIndex={-1} to=\\"/habitid1\\" component={{...}} focusVisibleClassName=\\"Mui-focusVisible\\">
                                                                        <Link className=\\"MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button\\" onBlur={[Function]} onClick={[undefined]} onFocus={[Function]} onKeyDown={[Function]} onKeyUp={[Function]} onMouseDown={[Function]} onMouseLeave={[Function]} onMouseUp={[Function]} onDragLeave={[Function]} onTouchEnd={[Function]} onTouchMove={[Function]} onTouchStart={[Function]} tabIndex={-1} role=\\"menuitem\\" aria-disabled={false} to=\\"/habitid1\\">
                                                                          <LinkAnchor className=\\"MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button\\" onBlur={[Function]} onClick={[undefined]} onFocus={[Function]} onKeyDown={[Function]} onKeyUp={[Function]} onMouseDown={[Function]} onMouseLeave={[Function]} onMouseUp={[Function]} onDragLeave={[Function]} onTouchEnd={[Function]} onTouchMove={[Function]} onTouchStart={[Function]} tabIndex={-1} role=\\"menuitem\\" aria-disabled={false} href=\\"/habitid1\\" navigate={[Function: navigate]}>
                                                                            <a className=\\"MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button\\" onBlur={[Function]} onFocus={[Function]} onKeyDown={[Function]} onKeyUp={[Function]} onMouseDown={[Function]} onMouseLeave={[Function]} onMouseUp={[Function]} onDragLeave={[Function]} onTouchEnd={[Function]} onTouchMove={[Function]} onTouchStart={[Function]} tabIndex={-1} role=\\"menuitem\\" aria-disabled={false} href=\\"/habitid1\\" onClick={[Function: onClick]}>
                                                                              <div className=\\"HabitItemMenuContent-root-59\\">
                                                                                <MaterialDesignIcon className=\\"HabitItemMenuContent-icon-60\\" name=\\"pencil\\">
                                                                                  <span className=\\"material-icon mdi mdi-pencil HabitItemMenuContent-icon-60\\" style={[undefined]} />
                                                                                </MaterialDesignIcon>
                                                                                <span className=\\"HabitItemMenuContent-text-61\\">
                                                                                  Edit
                                                                                </span>
                                                                              </div>
                                                                              <NoSsr>
                                                                                <WithStyles(undefined) center={false}>
                                                                                  <ForwardRef(TouchRipple) classes={{...}} center={false}>
                                                                                    <span className=\\"MuiTouchRipple-root\\">
                                                                                      <TransitionGroup component={{...}} exit={true} childFactory={[Function: childFactory]} />
                                                                                    </span>
                                                                                  </ForwardRef(TouchRipple)>
                                                                                </WithStyles(undefined)>
                                                                              </NoSsr>
                                                                            </a>
                                                                          </LinkAnchor>
                                                                        </Link>
                                                                      </ForwardRef(ButtonBase)>
                                                                    </WithStyles(ForwardRef(ButtonBase))>
                                                                  </ForwardRef(ListItem)>
                                                                </WithStyles(ForwardRef(ListItem))>
                                                              </ForwardRef(MenuItem)>
                                                            </WithStyles(ForwardRef(MenuItem))>
                                                          </HabitItemMenuContent>
                                                        </WithStyles(HabitItemMenuContent)>
                                                        <WithStyles(HabitItemMenuContent) icon=\\"pause\\" title=\\"Pause\\" action={[Function: onPause]}>
                                                          <HabitItemMenuContent classes={{...}} icon=\\"pause\\" title=\\"Pause\\" action={[Function: onPause]}>
                                                            <WithStyles(ForwardRef(MenuItem)) onClick={[Function: onPause]}>
                                                              <ForwardRef(MenuItem) classes={{...}} onClick={[Function: onPause]}>
                                                                <WithStyles(ForwardRef(ListItem)) button={true} role=\\"menuitem\\" tabIndex={-1} component=\\"li\\" selected={[undefined]} disableGutters={false} classes={{...}} className=\\"MuiMenuItem-root MuiMenuItem-gutters\\" onClick={[Function: onPause]}>
                                                                  <ForwardRef(ListItem) classes={{...}} button={true} role=\\"menuitem\\" tabIndex={-1} component=\\"li\\" selected={[undefined]} disableGutters={false} className=\\"MuiMenuItem-root MuiMenuItem-gutters\\" onClick={[Function: onPause]}>
                                                                    <WithStyles(ForwardRef(ButtonBase)) className=\\"MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button\\" disabled={false} role=\\"menuitem\\" tabIndex={-1} onClick={[Function: onPause]} component=\\"li\\" focusVisibleClassName=\\"Mui-focusVisible\\">
                                                                      <ForwardRef(ButtonBase) classes={{...}} className=\\"MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button\\" disabled={false} role=\\"menuitem\\" tabIndex={-1} onClick={[Function: onPause]} component=\\"li\\" focusVisibleClassName=\\"Mui-focusVisible\\">
                                                                        <li className=\\"MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button\\" onBlur={[Function]} onClick={[Function: onPause]} onFocus={[Function]} onKeyDown={[Function]} onKeyUp={[Function]} onMouseDown={[Function]} onMouseLeave={[Function]} onMouseUp={[Function]} onDragLeave={[Function]} onTouchEnd={[Function]} onTouchMove={[Function]} onTouchStart={[Function]} tabIndex={-1} role=\\"menuitem\\" aria-disabled={false}>
                                                                          <div className=\\"HabitItemMenuContent-root-59\\">
                                                                            <MaterialDesignIcon className=\\"HabitItemMenuContent-icon-60\\" name=\\"pause\\">
                                                                              <span className=\\"material-icon mdi mdi-pause HabitItemMenuContent-icon-60\\" style={[undefined]} />
                                                                            </MaterialDesignIcon>
                                                                            <span className=\\"HabitItemMenuContent-text-61\\">
                                                                              Pause
                                                                            </span>
                                                                          </div>
                                                                          <NoSsr>
                                                                            <WithStyles(undefined) center={false}>
                                                                              <ForwardRef(TouchRipple) classes={{...}} center={false}>
                                                                                <span className=\\"MuiTouchRipple-root\\">
                                                                                  <TransitionGroup component={{...}} exit={true} childFactory={[Function: childFactory]} />
                                                                                </span>
                                                                              </ForwardRef(TouchRipple)>
                                                                            </WithStyles(undefined)>
                                                                          </NoSsr>
                                                                        </li>
                                                                      </ForwardRef(ButtonBase)>
                                                                    </WithStyles(ForwardRef(ButtonBase))>
                                                                  </ForwardRef(ListItem)>
                                                                </WithStyles(ForwardRef(ListItem))>
                                                              </ForwardRef(MenuItem)>
                                                            </WithStyles(ForwardRef(MenuItem))>
                                                          </HabitItemMenuContent>
                                                        </WithStyles(HabitItemMenuContent)>
                                                        <WithStyles(HabitItemMenuContent) icon=\\"delete\\" title=\\"Delete\\" action={[Function: onDelete]}>
                                                          <HabitItemMenuContent classes={{...}} icon=\\"delete\\" title=\\"Delete\\" action={[Function: onDelete]}>
                                                            <WithStyles(ForwardRef(MenuItem)) onClick={[Function: onDelete]}>
                                                              <ForwardRef(MenuItem) classes={{...}} onClick={[Function: onDelete]}>
                                                                <WithStyles(ForwardRef(ListItem)) button={true} role=\\"menuitem\\" tabIndex={-1} component=\\"li\\" selected={[undefined]} disableGutters={false} classes={{...}} className=\\"MuiMenuItem-root MuiMenuItem-gutters\\" onClick={[Function: onDelete]}>
                                                                  <ForwardRef(ListItem) classes={{...}} button={true} role=\\"menuitem\\" tabIndex={-1} component=\\"li\\" selected={[undefined]} disableGutters={false} className=\\"MuiMenuItem-root MuiMenuItem-gutters\\" onClick={[Function: onDelete]}>
                                                                    <WithStyles(ForwardRef(ButtonBase)) className=\\"MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button\\" disabled={false} role=\\"menuitem\\" tabIndex={-1} onClick={[Function: onDelete]} component=\\"li\\" focusVisibleClassName=\\"Mui-focusVisible\\">
                                                                      <ForwardRef(ButtonBase) classes={{...}} className=\\"MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button\\" disabled={false} role=\\"menuitem\\" tabIndex={-1} onClick={[Function: onDelete]} component=\\"li\\" focusVisibleClassName=\\"Mui-focusVisible\\">
                                                                        <li className=\\"MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button\\" onBlur={[Function]} onClick={[Function: onDelete]} onFocus={[Function]} onKeyDown={[Function]} onKeyUp={[Function]} onMouseDown={[Function]} onMouseLeave={[Function]} onMouseUp={[Function]} onDragLeave={[Function]} onTouchEnd={[Function]} onTouchMove={[Function]} onTouchStart={[Function]} tabIndex={-1} role=\\"menuitem\\" aria-disabled={false}>
                                                                          <div className=\\"HabitItemMenuContent-root-59\\">
                                                                            <MaterialDesignIcon className=\\"HabitItemMenuContent-icon-60\\" name=\\"delete\\">
                                                                              <span className=\\"material-icon mdi mdi-delete HabitItemMenuContent-icon-60\\" style={[undefined]} />
                                                                            </MaterialDesignIcon>
                                                                            <span className=\\"HabitItemMenuContent-text-61\\">
                                                                              Delete
                                                                            </span>
                                                                          </div>
                                                                          <NoSsr>
                                                                            <WithStyles(undefined) center={false}>
                                                                              <ForwardRef(TouchRipple) classes={{...}} center={false}>
                                                                                <span className=\\"MuiTouchRipple-root\\">
                                                                                  <TransitionGroup component={{...}} exit={true} childFactory={[Function: childFactory]} />
                                                                                </span>
                                                                              </ForwardRef(TouchRipple)>
                                                                            </WithStyles(undefined)>
                                                                          </NoSsr>
                                                                        </li>
                                                                      </ForwardRef(ButtonBase)>
                                                                    </WithStyles(ForwardRef(ButtonBase))>
                                                                  </ForwardRef(ListItem)>
                                                                </WithStyles(ForwardRef(ListItem))>
                                                              </ForwardRef(MenuItem)>
                                                            </WithStyles(ForwardRef(MenuItem))>
                                                          </HabitItemMenuContent>
                                                        </WithStyles(HabitItemMenuContent)>
                                                      </ul>
                                                    </ForwardRef(List)>
                                                  </WithStyles(ForwardRef(List))>
                                                </ForwardRef(MenuList)>
                                              </div>
                                            </ForwardRef(Paper)>
                                          </WithStyles(ForwardRef(Paper))>
                                        </Transition>
                                      </ForwardRef(Grow)>
                                      <div tabIndex={0} data-test=\\"sentinelEnd\\" />
                                    </TrapFocus>
                                  </div>
                                </Portal>
                              </ForwardRef(Portal)>
                            </ForwardRef(Modal)>
                          </ForwardRef(Popover)>
                        </WithStyles(ForwardRef(Popover))>
                      </ForwardRef(Menu)>
                    </WithStyles(ForwardRef(Menu))>
                  </div>
                </HabitItemMenu>
              </Connect(HabitItemMenu)>
            </WithStyles(undefined)>
          </div>
        </div>
      </HabitItem>
    </Provider>
  </Router>
</MemoryRouter>"
`;
